lib/vector-buffers/src/variants/disk_v2/reader.rs

```diff
@@ -137,6 +136,32 @@ where
                 | ReaderError::PartialWrite
         )
     }
+
+    fn as_error_code(&self) -> &'static str {
+        match self {
+            ReaderError::Io { .. } => "io_error",
+            ReaderError::Deserialization { .. } => "deser_failed",
+            ReaderError::Checksum { .. } => "checksum_mismatch",
+            ReaderError::Decode { .. } => "decode_failed",
+            ReaderError::Incompatible { .. } => "incompatible_record_version",
+            ReaderError::PartialWrite => "partial_write",
+            ReaderError::EmptyRecord => "empty_record",
+        }
+    }
+
+    pub fn as_recoverable_error(&self) -> Option<BufferReadError> {
+        let error = self.to_string();
+        let error_code = self.as_error_code();
+
+        match self {
+            ReaderError::Io { .. } | ReaderError::EmptyRecord => None,
+            ReaderError::Deserialization { .. }
+            | ReaderError::Checksum { .. }
+            | ReaderError::Decode { .. }
+            | ReaderError::Incompatible { .. }
+            | ReaderError::PartialWrite => Some(BufferReadError { error_code, error }),
```

Comment:
Every error that _doesn't_ involve either an I/O error or an "empty record" is recoverable in this scenario.

I/O errors aren't recoverable because we don't know the extent of what the error implies, so shutting down is the only thing we can reasonably do there. Similarly, an "empty record" error is when a record reports that it contains 0 events total (`EventCount::event_count`) which is entirely nonsensical.

We technically check for records reporting an event count of zero on the write path, too.. so we should never seen this on the read path unless something is wildly wrong, hence why we've left it as unrecoverable, forcing Vector to abort to avoid further weirdness.